syntax = "proto3";

package handshake;

service HandShake {
  rpc SendTable (StackedTable) returns (ArrayResponse) {}
  rpc Aggregate (AggregateRequest) returns (AggregateResponse) {}
  rpc GetResults (GetResultsRequest) returns (StackedTable) {}
  rpc Shutdown( ShutdownRequest) returns (ShutdownResponse) {}
  rpc Hello (HelloRequest) returns (HelloResponse) {}  
}

message HelloRequest {
  string name = 1;
}

message HelloResponse {
  string reply = 1;
}
  
message ShutdownRequest {}

message ShutdownResponse {
    string status = 1;
}


message StackedTable {
    string doublename = 1;
    repeated double doublevalues = 2;
    string stringname = 3;
    repeated string stringvalues = 4;
}

message ArrayResponse {
  string message = 1;
}

message AggregateRequest {
    repeated int32 ports = 1;
}

message AggregateResponse {
    int32 return_code = 1;
}

message GetResultsRequest {
    int32 which_results = 1;
}
